notebooks/stac-load-S2-deafrica.py

```diff
@@ -0,0 +1,154 @@
+#
+# All assets except SLC have the same configuration. SLC uses `uint8` rather than `uint16`.
+#
+# In the configuration, we also supply the aliases for each band. This means we can load data by band name rather than band number.
+
+# %%
+config = {
+    "s2_l2a": {
+        "assets": {
+            "*": {
+                "data_type": "uint16",
+                "nodata": 0, 
+                "unit": "1",
+            },
+            "SLC": {
+                "data_type": "uint8",
+                "nodata": 0, 
+                "unit": "1",
+            },
+        },
+        "aliases": {
+            "costal_aerosol": "B01",
+            "blue": "B02",
+            "green": "B03",
+            "red": "B04",
+            "red_edge_1": "B05",
+            "red_edge_2": "B06",
+            "red_edge_3": "B07",
+            "nir": "B08",
+            "nir_narrow": "B08A",
+            "water_vapour": "B09",
+            "swir_1": "B11",
+            "swir_2": "B12",
+            "mask": "SLC",
+            "aerosol_optical_thickness": "AOT",
+            "scene_average_water_vapour": "WVP",
+        }
+    }
+}
+
+# %% [markdown]
+# ## Connect to the Digital Earth Africa STAC catalog
+
+# %%
+# Open the stac catalogue
+deafrica_stac_address = 'https://explorer.digitalearth.africa/stac'
+catalog = Client.open(deafrica_stac_address)
+
+# %% [markdown]
+# ## Find STAC Items to Load
+
+# %%
+# Construct a bounding box to search over
+# [xmin, ymin, xmax, ymax] in latitude and longitude
+bbox = [37.76, 12.49, 37.77, 12.50]
+
+# Construct a time range to search over
+start_date = '2020-09-01'
+end_date = '2020-12-01'
+timerange = f'{start_date}/{end_date}'
+
+# Choose the product/s to load
+products = ['s2_l2a']
+
+# %%
+# Identify all data matching the above:
+query = catalog.search(
+    bbox=bbox,
+    collections=products,
+    datetime=timerange
+)
+
+items = list(query.get_items())
+print(f"Found: {len(items):d} datasets")
+
+# %% [markdown]
+# ## Construct Dask Dataset
+#
+# In this step, we specify the desired coordinate system, resolution (here 20m), and bands to load. We also pass the bounding box to the `stac_load` function to only load the requested data.
+
+# %%
+crs = 'EPSG:6933'
+resolution = 20
+
+ds = stac_load(
+    items,
+    bands=("red", "green", "blue", "nir"),
+    crs=crs,
+    resolution=resolution,
+    chunks={},
+    groupby="solar_day",
+    stac_cfg=config,
+    bbox=bbox,
+)
+
+
+# %%
+# View the Xarray Dataset
+ds
+
+# %% [markdown]
+# ## Load the data into memory
+#
+# Digital Earth Africa data is stored on S3 in Cape Town, Africa. To load the data, we must use a rasterio Env configured with the appropriate AWS S3 endpoint.
+
+# %%
+# Load into memory
+with rasterio.Env(AWS_S3_ENDPOINT='s3.af-south-1.amazonaws.com', AWS_NO_SIGN_REQUEST='YES'):
+    ds_loaded = ds.load()
```

Comment:
This won't work with remote dask cluster.

The preferred way is to use `odc.stac.configure_rio`

https://odc-stac.readthedocs.io/en/latest/_api/odc.stac.configure_rio.html#odc.stac.configure_rio

```python
configure_rio(
   cloud_defaults=True, 
   aws={"aws_unsigned": True},
   AWS_S3_ENDPOINT='s3.af-south-1.amazonaws.com')
```

Then you won't need to load before computing ndvi.

Reply:
This is great to know about, thanks! Done